src/omni/embed.py

```diff
@@ -56,22 +56,25 @@ class Theme(Enum):
         blank = "blank"
 
     def __init__(
-        self, organization_name: str | None = None, embed_secret: str | None = None
+        self,
+        organization_name: str | None = None,
+        embed_secret: str | None = None,
+        vanity_domain: str | None = None,
     ):
         _organization_name = organization_name or OmniEnv.ORGANIZATION_NAME
-        if not _organization_name:
-            raise ValueError(
-                "organization_name is required if it is not configured in environment variables."
-            )
+        _vanity_domain = vanity_domain or OmniEnv.VANITY_DOMAIN
+        if not (_organization_name or _vanity_domain):
+            raise ValueError("'vanity_domain' or 'organization_name' are required.")
         _embed_secret = embed_secret or OmniEnv.EMBED_SECRET
         if not _embed_secret:
             raise ValueError(
                 "embed_secret is required if it is not configured in environment variables."
             )
 
-        self.embed_login_url = (
-            f"https://{_organization_name}.embed-omniapp.co/embed/login"
+        embed_host = (
```

Comment:
```suggestion
        embed_host = _vanity_domain or f"{_organization_name}.embed-omniapp.co"
```

Real note here is we need to use  `_vanity_domain` instead of `vanity_domain` but also shortened up the code.